Updating UKIT009

martinaxgloria opened the pull request:
This PR contains:

- @MeliTaliani commit with MKIT009 update;
- the refactor of the `post-flashing operations` after jetpack installation: now, this information is stored in the dedicated pages, so the Jetpack installation page is shorter and clearer now
- the procedure to setup the Orin NX to be used with FRAMOS-IMX415 cameras

What is missing:

- [x] software instruction about the UKIT009 update
- [x] review of the kinematics of the new UKIT009

I'll keep the PR in draft until the previous points are addressed.

For the time being, I generated the documentation on my fork to see the output:

- https://martinaxgloria.github.io/documentation/

cc @Nicogene @pattacini

martinaxgloria commented:
I don't know if it is the cause of the CI failure, but I'm using a link whose target is not already present in the documentation (i.e. I wrote that in iKin the new iCub version will be available since release v2024.11.0 with the reference to the markdown that not exist yet). When I do `mkdocs serve` to build the page locally, I retrieved only a warning and the page is built anyway.

martinaxgloria commented:
I removed the reference to the page that doesn't exist yet and I will add that once we have the release v2024.11.0

pattacini commented:
Thanks heaps for the contributions!
I'll be reviewing the PR in the next few days.

pattacini commented on docs/icub_operating_systems/icubos/install-cmake.md:
```diff
@@ -0,0 +1,27 @@
+# Install the latest CMake version (MANDATORY only for Ubuntu 18.04)
+The latest Robotology superbuild version requires CMake >= 3.12, but Ubuntu 18.04 only provides CMake 3.10, so it needs to be upgraded.
+
```
Clearly, Ubuntu 18.04 is far away in the past and we no longer need to address it.

That said, it could be still convenient to keep this info about installing the latest CMake wrt the version shipped with a specific Ubuntu distro.

If this is the case, then I would only rephrase the intro saying when these instructions may apply.

pattacini commented on docs/icub_operating_systems/icubos/jetpack.md:
```diff
@@ -130,100 +134,13 @@ sudo apt install -f
+- [Install CUDA libraries](./install-cuda-libraries.md)
+- [Install librealsense](./install-librealsense.md)
+- [Install jtop](./install-jtop.md)
+- [Install CMake (for Ubuntu 18.04)](./install-cmake.md)
```
Same as above.

pattacini commented on mkdocs.yml:
```diff
@@ -214,6 +214,11 @@ nav:
         - Network: ergocub_operating_systems/network.md
         - ergocub-head:
           - Jetpack installation: icub_operating_systems/icubos/jetpack.md
+          - Post-flashing operations:
+            - Install CUDA libraries: icub_operating_systems/icubos/install-cuda-libraries.md
+            - Install librealsense: icub_operating_systems/icubos/install-librealsense.md
+            - Install jtop: icub_operating_systems/icubos/install-jtop.md
+            - Install CMake (Ubuntu 18.04 only): icub_operating_systems/icubos/install-cmake.md
```
Too many details.
Please, lump all these pages together into one single page to be referenced for the post-flashing operations.
Rule out the part on CMake as per other comments and link it from somewhere else.

pattacini commented on mkdocs.yml:
```diff
@@ -245,6 +250,12 @@ nav:
         - OS images download: icub_operating_systems/download.md
         - iCub Setup startup procedure: icub_operating_systems/startup-procedure.md
         - NVIDIA JetPack installation: icub_operating_systems/icubos/jetpack.md
+        - Post-flashing operations:
+          - Install CUDA libraries: icub_operating_systems/icubos/install-cuda-libraries.md
+          - Install librealsense: icub_operating_systems/icubos/install-librealsense.md
+          - Install jtop: icub_operating_systems/icubos/install-jtop.md
+          - Install CMake (Ubuntu 18.04 only): icub_operating_systems/icubos/install-cmake.md
```
Same here.

pattacini reviewed (changes requested):
Hi @martinaxgloria 

Well done so far but I reckon we have to streamline a bit this PR, especially in terms of number of pages for the post-flashing procedure.

Also, we have to fix these warnings:
```console
INFO    -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
             - icub_operating_systems/icubos/create-backup-image.md
INFO    -  Doc file 'icub_operating_systems/icubos/create-backup-image.md' contains a link '#jetpack-setup-on-the-host', but there is no such anchor on this page.
INFO    -  Doc file 'icub_operating_systems/icubos/create-backup-image.md' contains a link '#booting-the-nvidia-jetson-board-in-recovery-mode', but there is no such anchor on this page.
```

martinaxgloria commented:
Hi @pattacini, thanks for the review! 
About:

> Ubuntu 18.04 is far away in the past and I guess we no longer need to address it.
> 
> That said, it could be still convenient to keep this info about installing the latest CMake wrt the version shipped with a specific Ubuntu distro.
> 
> If this is the case, then I would only rephrase the intro saying when these instructions may apply and why.

you're right, I kept this info since, if I remember well, the Xavier NX is Ubuntu 18.04 based yet. But this board will become outdated with the UKIT009 update, so we could think about removing it from the documentation and refactoring the CMake installation instructions depending on the OS version.

About the number of pages for the post-flashing operations, instead, talking with @Nicogene we agreed that the `Jetpack installation` page was too long and that it'll be better to remove the post-flashing operations from that page and re-organize this section into small dedicated pages. Tomorrow or in the next few days we can align f2f on that

pattacini commented:
>About the number of pages for the post-flashing operations, instead, talking with @Nicogene we agreed that the Jetpack installation page was too long and that it'll be better to remove the post-flashing operations from that page and re-organize this section into small dedicated pages. Tomorrow or in the next few days we can align f2f on that

Disagree with that 😉 
- Honestly I don't like the final rendering that is split into so many pages.
- Also, an intro page for the post-flashing is missing, which is, I think, the only page we may refer in the navigation section.

martinaxgloria commented:
In https://github.com/icub-tech-iit/documentation/pull/370/commits/33e7dc75d1bacbc973337663fcc487e1e9643e0b I applied the suggested changes after the f2f chat we had yesterday (cc @Nicogene @pattacini). In particular, I:

- created an introduction page for the `post-flashing operations` with a table of context in which are listed the available optional procedures (I removed the "optional" from the title of each page)
- removed from the navigation tree those pages and I added only the introduction page for both iCub and ergoCub installation 
- added an admonition with a reference to the jetpack installation page: in this way, each page can self-exists 
- removed the procedures for the creation of the backup image (not working) and cmake installation for Ubuntu 18.04 (not needed)

martinaxgloria commented:
I regenerated the pages if you want to see the output now

pattacini commented on docs/icub_operating_systems/icubos/jetpack.md:
```diff
@@ -95,6 +95,10 @@ The board is in recovery mode if it appears something like this
 
+!!! warning
+
+    If you want to configure your `Jetson Orin NX` to be used with `FRAMOS-IMX415` cameras, please refer to the [specific procedure](./setup-framos-imx415.md) from here on out.
```
```suggestion
    If you want to configure your `Jetson Orin NX` to be used with `FRAMOS-IMX415` cameras, please refer to the [specific procedure](./setup-framos-imx415.md) from hereinafter.
```

pattacini commented on docs/icub_versions/index.md:
```diff
@@ -17,7 +17,7 @@
 | v2.8   | v2.7 with beam-splitter in the eye for hosting event driven and frame based sensors | |
 | v2.9   | **v2.7** with hands mk5 and forearm mk2 | |   
-| v2.10  | **v2.7** with hands mk5, forearm mk2, new 4k basler cameras, nvidia xavier and intel i7 11th on com express type 10 module (UKIT009)  | |   
+| v2.10  | **v2.7** with hands mk5, forearm mk2, new 4K FRAMOS-IMX145 cameras, NVidia Orin NX [(UKIT009)](../upgrade_kits/head_4k/support.md) and intel i7 11th on com express type 10 module in the backpack | |   
```
```suggestion
| v2.10  | **v2.7** with the 4K FRAMOS-IMX145 cameras, NVidia Orin NX [(UKIT009)](../upgrade_kits/head_4k/support.md) and intel i7 11th on com express type 10 module in the backpack | |   
```
We will never mount ergoCub forearms + hands on iCub.
After merging, I'll remove v2.9.

cc @Nicogene @maggia80 @Fabrizio69

pattacini reviewed (changes requested):
Please, consider these further notes.

martinaxgloria commented:
Hi @pattacini,

here https://github.com/icub-tech-iit/documentation/pull/370/commits/40ff57e5e91d04f90c8c9d9db2afcabaf0a9620a you can find the changes you suggested. I added the CMake installation instructions again but for Ubuntu 20.04 since it is required for yarp v3.10.0 and we had to follow this procedure on the laptop of iCubGenova11 which has Ubuntu 20.04 yet

Please, let me know if it's ok now, thanks!

pattacini reviewed (approved):
Thanks heaps!
Merging.
